week3/xor/jasonkuhrt/Main.hs

```diff
@@ -0,0 +1,67 @@
+{- README
+
+-- XOR Encryption
+
+-- Spec
+
+Write a program that encrypts a file by XORing it with a key. For instance:
+
+     Key  "XYZ"
+Contents  "abracadabra"
+     XOR   XYZXYZXYZXY
+       =   9;(9:;<88*8
+
+* It should accept the file name and the key as command-line arguments
+* Use bytestring to read the file
+* Encode the key as UTF8 (with encodeUtf8)
+* Overwrite the file with encryped variant
+
+-- Notes
+
+Due to the way XOR works, encrypting an already encrypted file will decrypt
+it. Learn more about XOR Ciphers at en.wikipedia.org/wiki/xor_cipher.
+-}
+
+module Main where
+
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as B
+import qualified Data.Bits as Bits
+import qualified Data.ByteString.Char8 as BC8
+
+
+
+{- | Encrypt a string via key+XOR+key. -}
+encrypt :: String -> ByteString -> ByteString
+encrypt key bs =
+  B.pack (B.zipWith Bits.xor (BC8.pack cycledKey) bs)
+  where
+  cycledKey = take (B.length bs) (cycle key)
```

Comment:
I have a concern about this. In unicode a character is one _or more_ code points. If we’re cycling a string to be the length of a bytestring, its not going to work because the length of the bytestring !== count of characters. In actuality cycling by the length of the bytestring might result in a string that is longer than the bytestring were it converted to a string. Am I mistaken?